packages/Telephony/Actions/FixedLengthBatchInput.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    /// <summary>
+    /// Aggregates input until it matches a regex pattern and then stores the result in an output property.
+    /// </summary>
+    public class FixedLengthBatchInput : RegexAggregatorInput
+    {
+        /// <summary>
+        /// Class identifier.
+        /// </summary>
+        [JsonProperty("$kind")]
+        public new const string Kind = "Microsoft.Telephony.FixedLengthBatchInput";
+
+        private int _batchLength;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FixedLengthBatchInput"/> class.
+        /// </summary>
+        /// <param name="regexAggregatorInput">Implementation for aggregatingInput.</param>
+        /// <param name="sourceFilePath">Optional, source file full path.</param>
+        /// <param name="sourceLineNumber">Optional, line number in source file.</param>
+        [JsonConstructor]
+        public FixedLengthBatchInput([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        /// <summary>
+        /// Gets or sets the character that will be used to signal that the batch of input is complete.
+        /// </summary>
+        [JsonProperty("batchLength")]
+        public int BatchLength
+        {
+            get
+            {
+                return _batchLength;
+            }
+
+            set
+            {
+                TerminationConditionRegexPattern = "(.){" + value + "}";
+                _batchLength = value;
```

Comment:
This test will fail in Webchat:

- Bot : Enter your 4 digit account number 
- User: 1
- User: 2
- User: 3
- User: 45
- Bot: You entered 12345

Telephony wont allow multi-char DTMF.

Reply:
We could lock this down to DTMF input only for aggregation with just a couple more lines. I wasn't sure that was necessary. The webchat test works exactly how you just said. I'm not sure that's a problem. If the user says their pin instead of types it via telephony, it works right now as well.

Reply:
Hmmm... I would have expected if batchLength is 4, I would get back a 4 char string not 5.

Reply:
Oh, I see, I misread this on the first pass. Let me make a note to chew on this one.